kratos/containers/csr_matrix.h

```diff
@@ -0,0 +1,587 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Riccardo Rossi
+//
+
+#if !defined(KRATOS_CSR_MATRIX_H_INCLUDED )
+#define  KRATOS_CSR_MATRIX_H_INCLUDED
+
+
+// System includes
+#include <iostream>
+#include "containers/sparse_contiguous_row_graph.h"
+#include "containers/system_vector.h"
+#include "utilities/parallel_utilities.h"
+#include "utilities/atomic_utilities.h"
+#include "includes/key_hash.h"
+
+// Project includes
+#include "includes/define.h"
+
+
+namespace Kratos
+{
+///@addtogroup ApplicationNameApplication
+///@{
+
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+/// This function implements "serial" CSR matrix, including capabilities for FEM assembly
+template< class TDataType=double, class TIndexType=std::size_t>
+class CsrMatrix
+{
+public:
+    ///@name Type Definitions
+    ///@{
+    typedef TIndexType IndexType;
+    typedef std::unordered_map<std::pair<IndexType, IndexType>,
+                          double,
+                          PairHasher<IndexType, IndexType>,
+                          PairComparor<IndexType, IndexType>
+                          > MatrixMapType;
+
+    /// Pointer definition of CsrMatrix
+    KRATOS_CLASS_POINTER_DEFINITION(CsrMatrix);
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+    CsrMatrix()
+    {
+    }
+
+    /// Default constructor.
+    template<class TGraphType>
+    CsrMatrix(const TGraphType& rSparseGraph)
+    {
+        rSparseGraph.ExportCSRArrays(mRowIndices,mColIndices);
+        mNrows = size1();
+
+        ComputeColSize();
+
+        //initialize mValuesVector to zero
+        mValuesVector.resize(mColIndices.size(),false);
+        SetValue(0.0);
+    }
+
+    /// Destructor.
+    virtual ~CsrMatrix(){}
+
+    /// Assignment operator. TODO: decide if we do want to allow it
+    CsrMatrix& operator=(CsrMatrix const& rOther)=delete;
+    // {
+    //     this->AddEntries(rOther.GetGraph());
+    //     return *this;
+    // }
+
+    ///@}
+    ///@name Operators
+    ///@{
+    void Clear()
+    {
+        mRowIndices.clear();
+        mColIndices.clear();
+        mValuesVector.clear();
+    }
+
+    void SetValue(const TDataType value)
```

Comment:
I'd call this `SetValueAllEntries` or something of the sort. Maybe it makes also sense to have a `SetToZero` method that calls this one with a 0.0 `value`.

Reply:
(I don't know if we need/want to match any already existent API or we can create our own).

Reply:
Fine then. In any case it might have sense to create a new `SetToZero` in both matrices.